### DISCUSSION TITLE
Question: How to use migrations and factories from a different package

### DISCUSSION THREAD

**Discussion by vishytk** [orchestral/testbench#291] [Q&A]
I am creating different packages for each function of an app. 

For example: 
1. admin is one function responsible for managing users and roles
2. student (belongs to student role)

I have created "**admin**" package and now working on creating the **"student"** package. In the **"student"** package tests/TestCase.php setup() i am creating a user with role student. 

```php
    protected function setUp(): void
    {
        parent::setUp();

        $this->user = UserFactory::new()->create([
            'username' => 'raju',
            'firstname' => 'Babruvahana',
            'lastname' => 'Arjuna',
            'role_id' => RoleFactory::new([
                'role' => 'student'
            ])
        ]);

        $this->headers = ['Accept' => 'application/json'];
    }
```

Now, the factories, migrations for users and roles belong to **"admin"** package and in order to create a student user, I have to copy the factories and migrations files from "**admin**" package to "**student**" package database/folders.

Is there a way to use these files without copying them over ?

**Comment by crynobone** [comment]
If the factories files are autoloaded via composer and not only available from `require-dev` you can just require it.

**Comment by vishytk** [comment]
I updated the namespace in factories file from:
namespace Database\Factories; to
namespace Vendor\Package\Database\Factories;

how does one use the migrations file(s) without copying ?
